cpp/src/digitize/digitize.cu

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "digitize.hpp"
+#include <cudf/digitize.hpp>
+#include <cudf/column/column_factories.hpp>
+#include <cudf/column/column_device_view.cuh>
+#include <cudf/table/row_operators.cuh>
+#include <cudf/utilities/traits.hpp>
+#include <cudf/utilities/type_dispatcher.hpp>
+#include <utilities/error_utils.hpp>
+#include <thrust/binary_search.h>
+
+namespace cudf {
+
+namespace {
+
+// Initialize column with sequential values starting from 0
+auto make_sequence(size_type num_rows, rmm::mr::device_memory_resource* mr, cudaStream_t stream) {
```

Comment:
No need for this if using counting iterators.

Reply:
Done